cpp/src/arrow/acero/hash_aggregate_test.cc

```diff
@@ -4206,6 +4206,235 @@ TEST_P(GroupBy, MinMaxWithNewGroupsInChunkedArray) {
                     /*verbose=*/true);
 }
 
+TEST_P(GroupBy, FirstLastBasicTypes) {
+  std::vector<std::shared_ptr<DataType>> types;
+  types.insert(types.end(), boolean());
+  types.insert(types.end(), NumericTypes().begin(), NumericTypes().end());
+  types.insert(types.end(), TemporalTypes().begin(), TemporalTypes().end());
+
+  const std::vector<std::string> default_table = {R"([
+    [1,    1],
+    [null, 1]
+])",
+                                                  R"([
+    [0,    2],
+    [null, 3],
+    [3,    4],
+    [5,    4],
+    [4,    null],
+    [3,    1],
+    [0,    2]
+])",
+                                                  R"([
+    [0,    2],
+    [1,    null],
+    [null, 3]
+])"};
+
+  const std::string default_expected =
+      R"([
+    [1,    1,    3,    null,   null],
```

Comment:
@icexelloss ping

Reply:
Good point. Now I think of this, the behavior of `skip_nulls=false` you described makes more sense. I will update this.

Reply:
As I remember, they mimic dplyr's `na.rm`:

> If na.rm is FALSE an NA value in any of the arguments will cause a value of NA to be returned, otherwise NA values are ignored. 

https://rdrr.io/r/base/Extremes.html

dplyr appears to interpret `na_rm` differently for `slice` (~= first, last):

> Should missing values in order_by be removed from the result? If FALSE, NA values are sorted to the end (like in [arrange()](https://dplyr.tidyverse.org/reference/arrange.html)), so they will only be included if there are insufficient non-missing values to reach n/prop.

https://dplyr.tidyverse.org/reference/slice.html

Reply:
I think this is closer to [`first`/`last`](https://dplyr.tidyverse.org/reference/nth.html) which states:

> na_rm : Should missing values in x be removed before extracting the value?

Reply:
Seems like we're all on the same page here.

Reply:
I'm going to unsubscribe for my inbox's sake, ping me again if something comes up